service/migration_manager.cc

```diff
@@ -1058,13 +1058,21 @@ static future<schema_ptr> get_schema_definition(table_schema_version v, locator:
             // with TTL (refresh TTL in case column mapping already existed prior to that).
             auto us = s.unfreeze(db::schema_ctxt(proxy));
             // if this is a view - sanity check that its schema doesn't need fixing.
+            schema_ptr base_schema;
             if (us->is_view()) {
                 auto& db = proxy.local().local_db();
-                schema_ptr base_schema = db.find_schema(us->view_info()->base_id());
+                base_schema = db.find_schema(us->view_info()->base_id());
                 db::schema_tables::check_no_legacy_secondary_index_mv_schema(db, view_ptr(us), base_schema);
             }
-            return db::schema_tables::store_column_mapping(proxy, us, true).then([us] {
-                return frozen_schema{us};
+            return db::schema_tables::store_column_mapping(proxy, us, true).then([us, base_schema] -> base_and_view_schemas {
+                if (us->is_view()) {
+                    if (!us->view_info()->base_info()) {
+                        us->view_info()->set_base_info(us->view_info()->make_base_dependent_view_info(*base_schema));
+                    }
```

Comment:
us is temporary and going away, it's probably unnecessary to set base info here?

Reply:
Right, we're freezing it anyway and we have the base schema from elsewhere. I'm removing it in the rebase